BusinessLogic_BookAPI/Services/LibraryService.cs

```diff
@@ -0,0 +1,341 @@
+﻿namespace BusinessLogic_BookAPI.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using BusinessLogic_BookAPI.Models;
+
+    /// <summary>
+    /// Library manager
+    /// </summary>
+    /// <seealso cref="BusinessLogic_BookAPI.Services.ILibraryService" />
+    public class LibraryService : ILibraryService
+    {
+        #region Initializing
+
+        /// <summary>
+        /// List of books
+        /// </summary>
+        private List<Book> _books = new List<Book>();
+
+        /// <summary>
+        /// List of authors
+        /// </summary>
+        private List<Author> _authors = new List<Author>();
+
+        /// <summary>
+        /// List of genres
+        /// </summary>
+        private List<Genre> _genres = new List<Genre>();
+
+        /// <summary>
+        /// Sorted set of book-genre link
+        /// </summary>
+        private SortedSet<BookGenrePair> _bookGenrePair
+            = new SortedSet<BookGenrePair>();
+
+        /// <summary>
+        /// Sorted set of book-author pair
+        /// </summary>
+        private SortedSet<BookAuthorPair> _bookAuthorPair
+            = new SortedSet<BookAuthorPair>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LibraryService"/> class.
+        /// </summary>
+        public LibraryService(IDataProvider data)
+        {
+            _books = data.SetBooks().ToList();
+            _authors = data.SetAuthors().ToList();
+            _genres = data.SetGenres().ToList();
+            _bookAuthorPair = new SortedSet<BookAuthorPair>(data.SetBooksAuthors());
+            _bookGenrePair = new SortedSet<BookGenrePair>(data.SetBooksGenres());
+        }
+
+        #endregion
+
+        #region Creating
+
+
+        /// <summary>
+        /// Creates and adds to list an author.
+        /// </summary>
+        /// <param name="author">The author.</param>
+        /// <returns>
+        /// Created author.
+        /// </returns>
+        public Author CreateAuthor(Author author)
```

Comment:
About naming: as I understand, you don't create author hear. You add author to your base here